cmake/SetupGitHooks.cmake

```diff
@@ -1,34 +1,46 @@
 # Distributed under the MIT License.
 # See LICENSE.txt for details.
 
-# Check that the source dir is writable. If it is we set up git hooks, if not
-# then there probably won't be any commits anyway...
-EXECUTE_PROCESS(COMMAND test -w ${CMAKE_SOURCE_DIR}
+# Allow disabling the Git hooks because if you are running the code in a
+# container the host may not have all the right things installed. For example,
+# if you are using python2 in the container, the host may not have python2
+# installed and we should instead be setting up hooks from the host or an
+# environment suitably similar to where the commits will be made.
+option(
+  USE_GIT_HOOKS
+  "Set up the git hooks for sanity checks."
+  ON)
+
+if(USE_GIT_HOOKS)
```

Comment:
How about

```cmake
if(NOT USE_GIT_HOOKS)
  return()
endif()
```

Reply:
Is that the recommended way? This isn't a function so I don't think you can return from inside it. I thought CMake's include was similar to C/C++ include where it's a verbatim insertion of the text

Reply:
No idea what's "recommended", or if such things as "best-practices" even exist with cmake :)

Reply:
okay, yes there are. Since the pattern here is what we currently do everywhere, why not leave it?

Reply:
fine with me